google-basemap/index.ts

```diff
@@ -0,0 +1,133 @@
+import './style.css';
+import {GoogleMapsOverlay as DeckOverlay} from '@deck.gl/google-maps';
+import {colorCategories} from '@deck.gl/carto';
+import {VectorTileLayer, vectorQuerySource} from '@deck.gl/carto';
+
+const GOOGLE_MAPS_API_KEY = import.meta.env.VITE_GOOGLE_API_KEY;
+const GOOGLE_MAP_ID = '';
+const GOOGLE_MAPS_API_URL = `https://maps.googleapis.com/maps/api/js?key=${GOOGLE_MAPS_API_KEY}&v=beta&map_ids=${GOOGLE_MAP_ID}`;
+const apiBaseUrl = import.meta.env.VITE_API_BASE_URL;
+const accessToken = import.meta.env.VITE_API_ACCESS_TOKEN;
+const connectionName = import.meta.env.VITE_API_CONNECTION_NAME;
+const cartoConfig = {apiBaseUrl, accessToken, connectionName};
+const SQL_QUERY = `SELECT geom, country_name, continent_name, pop_2015 FROM cartobq.public_account.world_population_2015 WHERE CAST(pop_2015 as INT64) between CAST(@min as INT64) and CAST(@max as INT64) ORDER BY pop_2015 DESC`;
+
+let map = null;
+let overlay = null;
+
+function loadScript(url) {
+  const script = document.createElement('script');
+  script.type = 'text/javascript';
+  script.src = url;
+  const head = document.querySelector('head');
+  // @ts-ignore
+  head.appendChild(script);
+  return new Promise(resolve => {
+    script.onload = resolve;
+  });
+}
+
+function setTooltip({x, y, object}) {
+  const tooltip = document.getElementById('tooltip');
+  if (object && tooltip) {
+    tooltip.style.display = 'block';
+    tooltip.style.left = x + 'px';
+    tooltip.style.top = y + 'px';
+    tooltip.innerHTML = `<div style="padding: 8px">Continent: ${object.properties.continent_name}</br>Country: ${object.properties.country_name}</br>Population: ${object.properties.pop_2015}</div>`;
+    console.log(x, y, object);
+  } else if (tooltip) {
+    tooltip.style.display = 'none';
+  }
+}
+
+function getColorConfiguration() {
```

Comment:
No need to wrap in function, just do `const getFillColor = colorCategories({...`

Reply:
Done